### PR TITLE
Do not throw when parsing a null repeated field

### PR DESCRIPTION
**Pull request by jlabanca**
MessageTypeAdapter throws an IllegalArgumentException when attempting to parse a repeated field that is set to null in a JSON string.  Ideally, the null field would never have been included in the JSON to begin with, but the source may not be as smart as Wire.  MessageTypeAdapter should handle it anyway.

### CODE REVIEW

**Comment by danrice-square** [comment]
LGTM

File: wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.java [hunk 1]
```diff
@@ -136,8 +136,10 @@ private JsonElement parse(JsonReader in) {
   private Object parseValue(Label label, Type valueType, JsonElement valueElement) {
     if (label.isRepeated()) {
       List<Object> valueList = new ArrayList<>();
-      for (JsonElement element : valueElement.getAsJsonArray()) {
-        valueList.add(readJson(valueType, element));
+      if (!valueElement.isJsonNull()) {
```
Comment (JakeWharton) [inline wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.java]: Maybe flip this check and early return `Collections.emptyList()`?

Comment (JakeWharton) [inline wire-gson-support/src/main/java/com/squareup/wire/MessageTypeAdapter.java]: Avoids the allocation.

**Comment by JakeWharton** [comment]
Looks good

**Pull request merged**